CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -422,12 +422,36 @@ static function X2DataTemplate CreateInfiltrationActivityProxyReward ()
 
 static function string GetInfiltrationActionPreview (XComGameState_Reward RewardState)
 {
-	return GetInfiltrationTemplateFromReward(RewardState).ActionRewardDisplayName;
+	local XComGameState_MissionSiteInfiltration MissionState;
+	local XComGameState_Activity ActivityState;
+	local XComGameStateHistory History;
+	local StateObjectReference RewardRef;
+	local string strRewards;
+	
+	History = `XCOMHISTORY;
+	ActivityState = class'XComGameState_Activity'.static.GetActivityFromSecondaryObjectID(RewardState.RewardObjectReference.ObjectID);
+	MissionState = XComGameState_MissionSiteInfiltration(History.GetGameStateForObjectID(ActivityState.PrimaryObjectRef.ObjectID));
+
+	strRewards = "";
+
+	foreach MissionState.Rewards(RewardRef)
+	{
+		if (RewardRef != MissionState.Rewards[0])
+		{
+			strRewards = strRewards $ ", ";
+		}
+
+		RewardState = XComGameState_Reward(History.GetGameStateForObjectID(RewardRef.ObjectID));
+
+		strRewards = strRewards $ RewardState.GetRewardString();
+	}
+
+	return strRewards;
 }
 
 static function string GetInfiltrationActionDetails (XComGameState_Reward RewardState)
 {
-	return GetInfiltrationTemplateFromReward(RewardState).GetRewardDetailStringFn(class'XComGameState_Activity'.static.GetActivityFromSecondaryObjectID(RewardState.RewardObjectReference.ObjectID), RewardState);
```

Comment:
I think it would be better if you didn't short-circuit `GetRewardDetailStringFn` delegate on the template and moved your changes to the default implementation instead